src/main/scala/org/tessellation/aci/ACIRepository.scala

```diff
@@ -0,0 +1,45 @@
+package org.tessellation.aci
+
+import cats.data.OptionT
+import cats.effect.{Async, ContextShift}
+import doobie._
+import doobie.implicits._
+
+class ACIRepository[F[_]: Async: ContextShift](private val dbPath: String) {
+
+  private val xa = Transactor
+    .fromDriverManager[F]("org.sqlite.JDBC", s"jdbc:sqlite:$dbPath", "", "")
+
+  def findAciType(id: String): OptionT[F, ACIType] =
+    OptionT(
+      sql"""select id, address, definition, kryoRegistrationId from aci_type where id = $id"""
+        .query[ACIType]
+        .option
+        .transact(xa)
+    )
+
+  def findAciMappings(srcTypeId: String): F[List[(ACITypeMapping, ACIType)]] =
+    sql"""select m.src_type_id, m.dst_type_id, m.fn,
+                 t.id, t.address, t.definition
+          from aci_mapping m 
+          join aci_type t
+          on m.dst_type_id = t.id
+         where m.src_type_id = $srcTypeId"""
+      .query[(ACITypeMapping, ACIType)]
+      .to[List]
+      .transact(xa)
+
+  def saveAciType(aciType: ACIType): F[Int] =
```

Comment:
Here (or in the endpoint) we will want to do a validation step which checks if the type inherits from Omega

Reply:
is it of type A <: \Omega

Reply:
Ω